controllers/users.js

```diff
@@ -0,0 +1,174 @@
+const mongoose = require('mongoose')
+const Courses = require('../models/courses')
+const bcrypt = require('bcryptjs')
+const { Users, Roles } = require('../models/users')
+
+const userController = {};
+
+/**
+ * @param {Object} req 
+ * @param {Object} res 
+ * @param {Function} next 
+ * @description Fetches all users' data 
+ */
+userController.getAll = async (req, res) => {
+    try {
+        const allUsera = await Users.find()
+        const users = allUsera.filter(user => {
+            return user.is_deleted === false
+        })
+        res.json(users)
+    }
+    catch (err) {
+        res.send('Error' + err)
+    }
+}
+
+/**
+ * 
+ * @param {Object} req 
+ * @param {Object} res 
+ * @param {Function} next 
+ * @description Fetches one user
+ */
+userController.getOne = async (req, res) => {
+    try {
+        let UserId = mongoose.Types.ObjectId(req.params.id)
+        Users.exists(UserId, async (error, result) => {
```

Comment:
don't need to check if the userId exists or not here simply use the findById query and if there is no user it will return null